packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/challengeintegrationsteps.md

```diff
@@ -0,0 +1,50 @@
+### 1 - 4: Sign-in and Select Authenticator
+The challenge flow follows the same first four steps as the [enrollment flow](#integrate-sdk-for-authenticator-enrollment):
+
+* [Display the initial sign-on UI](#_1-display-the-initial-sign-on-ui).
+* [Start the sign-on flow](#_2-start-the-sign-on-flow).
+* [Authenticate the user credentials](#_3-authenticate-the-user-credentials).
+* [Display a list of authenticators available for enrollment](#_4-display-a-list-of-authenticators-available-for-enrollment).
+
+
+### 5: Display an OTP entry field
+
+The `.challengeAuthenticator` remediation type used for requesting a password is also used for one-time codes. A Google Authencator code request is represented by an authenticator with a `type` of `.app` and a `key` of `"google_otp"`.
+
+This version of `handleChallenge(remediation:)` includes Google Authenticator:
+
+
+```swift
+func handleChallenge(_ remediation: Remediation) {
+    guard let authenticator = remediation.authenticators.first else {
+        finish(with: .unexpectedAuthenticator)
+        return
+    }
+
+    let type = authenticator.type
+    switch type {
+    case .password:
+        remediation["credentials.passcode"]?.value = password
+        remediation.proceed()
+
+    case .app:
+        if authenticator.key == "google_otp" {
+            // Tell the UI to present a UI for requesting a one-time key.
+        }
+
+    default:
+        finish(with: .unexpectedAuthenticator)
+        return
+    }
+}
+```
+
+The `EnrollAuthenticatorView` from [the enrollment flow](#_5-display-the-shared-secret-qr-code-and-request-the-code) also works for the challenge case as the shared secret section is shown only if the data is in the remediation. The following screenshots show that view with a field for entering the OTP, and the same view display a message that's part of the remediation.
```

Comment:
```suggestion
The `EnrollAuthenticatorView` from [the enrollment flow](#_5-display-the-shared-secret-qr-code-and-request-the-code) also works for the challenge case as the shared secret section is shown only if the data is in the remediation. The following screenshots show that view with a field for entering the OTP, and the same view displays a message that's part of the remediation.
```

Reply:
First, I'm not sure how to comment on the flow diagram comments. I'll update the wording as appropriate. The diagrams themselves are draft version form which Alex will create final ones. I think all the text is copy/pasted from the .net flowcharts and then modified with the iOS specific info.

I'll do a revision for the next commit (likely late Monday).

For this item, thanks for catching the typo, the fix will be in next commit.